tools/circleci/build-and-deploy-images.py

```diff
@@ -0,0 +1,15 @@
+import yaml
+
+class SafeDict(dict):
+    def __missing__(self, key):
+        return '{' + key + '}'
+
+with open("Dockerfile.template") as f:
+    docker_template = f.read()
+
+with open("images.yml") as f:
+    custom_settings = yaml.safe_load(f)
+
+xenial_docker_image = docker_template.format_map(SafeDict(custom_settings["xenial"]))
+with open("images/xenial/Dockerfile", "w") as f:
+    f.write(xenial_docker_image)
```

Comment:
This script doesn't actually build any images, just Dockerfiles. (An image is what you get from `docker build` -- a lot like a [disk image](https://en.wikipedia.org/wiki/Disk_image). The "images/" name is kind of misleading; I copied it from Circle upstream; it's more like the source for a bunch of images, which for us is just Dockerfiles but could also include some more files.) That's why the name `generate-dockerfiles` is more accurate.

Also, let's have the script generate a Dockerfile for each item in `images.yml`, and not have any hard-coded release names. That's how we benefit from having a separate config file. :)

General Python style: when a file is meant to be run, like this one, rather than imported:
* it gets `#!/usr/bin/env python3` at the top
* it gets `chmod +x`
* its name leaves off the `.py`, which simultaneously makes it less not annoying to type as a command and also makes it impossible to import by accident.